### PR TITLE
Revert "Revert "[LT] Codegen ReuseNode for supported ops""

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77513

Summary: Fixed a XLC build failure by generating an always-return-false
default CanBeReused method.

This reverts commit 3cade9d454babc70f43ace1ce6481543104934b2.

### CODE REVIEW

File: torchgen/dest/lazy_ir.py [hunk 1]
```diff
@@ -129,6 +125,13 @@ def __call__(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> List[str]:
     def lowering_function(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> str:
         return ""
 
+    def can_be_reused_function(
+        self, f: Union[NativeFunctionsGroup, NativeFunction], node_ctor_args: str
+    ) -> str:
+        return f"""bool CanBeReused({node_ctor_args}) const {{
```
Comment (desertfire) [inline torchgen/dest/lazy_ir.py:131]: @JackCaoG This is where I added a default method implementation. Let me know if it looks ok.

Comment (JackCaoG) [reply torchgen/dest/lazy_ir.py:131]: this lgtm, thanks! Let me know if xla build still fail

**Review by alanwaketan — APPROVED** [review]
Stamped.

**Pull request closed without merging**